ipatests/test_integration/test_ipa_cert_fix.py

```diff
@@ -336,52 +336,88 @@ def install(cls, mh):
             setup_dns=False, extra_args=['--no-ntp']
         )
 
-    def test_renew_expired_cert_replica(self):
+    @pytest.fixture
+    def move_date_check_replication(self):
+        # move system date to expire certs
+        for host in self.master, self.replicas[0]:
+            move_date(host, 'stop', '+3years+1days')
+
+        yield
+
+        # move date back on replica and master
+        for host in self.master, self.replicas[0]:
+            move_date(host, 'start', '-3years-1days')
+
+    def test_renew_expired_cert_replica(self, move_date_check_replication):
         """Test renewal of certificates on replica with ipa-cert-fix
 
         This is to check that ipa-cert-fix renews the certificates
         on replica
 
         related: https://pagure.io/freeipa/issue/7885
         """
-        move_date(self.master, 'stop', '+3years+1days')
-
         # wait for cert expiry
         check_status(self.master, 8, "CA_UNREACHABLE")
 
         self.master.run_command(['ipa-cert-fix', '-v'], stdin_text='yes\n')
 
         check_status(self.master, 9, "MONITORING")
 
-        # move system date to expire cert on replica
-        move_date(self.replicas[0], 'stop', '+3years+1days')
+        # replica operations
+        # 'Server-Cert cert-pki-ca' cert will be in CA_UNREACHABLE state
+        cmd = self.replicas[0].run_command(
+            ['getcert', 'list',
+             '-d', paths.PKI_TOMCAT_ALIAS_DIR,
+             '-n', 'Server-Cert cert-pki-ca']
+        )
+        req_id = get_certmonger_fs_id(cmd.stdout_text)
+        tasks.wait_for_certmonger_status(
+            self.replicas[0], ('CA_UNREACHABLE'), req_id, timeout=600
+        )
+
+        # check that HTTP,LDAP,PKINIT are renewed and in MONITORING state
+        instance = realm_to_serverid(self.master.domain.realm)
+        dirsrv_cert = paths.ETC_DIRSRV_SLAPD_INSTANCE_TEMPLATE % instance
+        for cert in (paths.KDC_CERT, paths.HTTPD_CERT_FILE):
+            cmd = self.replicas[0].run_command(
+                ['getcert', 'list', '-f', cert]
+            )
+            req_id = get_certmonger_fs_id(cmd.stdout_text)
+            tasks.wait_for_certmonger_status(
+                self.replicas[0], ('MONITORING'), req_id, timeout=600
+            )
 
-        # RA agent cert will be expired and in CA_UNREACHABLE state
-        check_status(self.replicas[0], 1, "CA_UNREACHABLE")
+        cmd = self.replicas[0].run_command(
+            ['getcert', 'list', '-d', dirsrv_cert]
+        )
+        req_id = get_certmonger_fs_id(cmd.stdout_text)
+        tasks.wait_for_certmonger_status(
+            self.replicas[0], ('MONITORING'), req_id, timeout=600
+        )
```

Comment:
According to the man page:
```
       To  repair  certificates  across multiple CA servers, first ensure that
       LDAP replication is working across the topology.  Then run ipa-cert-fix
       on  one  CA  server.  Before running ipa-cert-fix on another CA server,
       trigger Certmonger renewals for shared certificates via  getcert-resub‐
       mit(1)  (on the other CA server).  This is to avoid unnecessary renewal
       of shared certificates.
```
I would try this scenario, ie call getcert resubmit on the replica for the RA cert + auditSigningCert cert-pki-ca, ocspSigningCert cert-pki-ca, subsystemCert cert-pki-ca which are shared certificates, then only call ipa-cert-fix.